src/main/java/org/spongepowered/api/entity/Entity.java

```diff
@@ -271,6 +271,24 @@ default boolean setLocationAndRotationSafely(Location<World> location, Vector3d
     boolean setTransform(Transform<World> transform);
 
     /**
+     * Sets the transformation of this entity using a safe location from
+     * {@link TeleportHelper#getSafeLocation(Location)}.
+     *
+     * @param transform The transform to set
+     * @return True if the transform was set successfully, or false if either
+     *     the transform couldn't be set as no safe location was found or
+     *     {@link org.spongepowered.api.event.entity.MoveEntityEvent.Teleport}
+     *     was cancelled
+     */
+    default boolean setTransformSafely(Transform<World> transform) {
+        checkNotNull(transform, "The transform cannot be null!");
```

Comment:
Sure, this works, but why not just `return setLocationAndRotationSafely(transform.getLocation(), transform.getRotation())` instead? It's one less place to change if we change the teleport helper, and it means we don't create an extra `Transform` object.

Reply:
Good points. Works for me, I'll make that change.